projects/api/src/controllers/authController.ts

```diff
@@ -1,12 +1,67 @@
-import {getConnection, getRepository} from 'typeorm'
-import {User} from '../entities'
+import { getRepository, Repository } from 'typeorm'
+import { User } from '../entities'
+import { IUser } from '../interfaces'
+import Boom, { Boom as BoomType } from '@hapi/boom'
+import bcrypt from 'bcrypt'
+import jwt from 'jsonwebtoken'
+import { secretKey } from '../constants/secret'
+import { DateTime } from 'luxon'
+
+/**
+ * This private method checks the user's credentials and generates a token
+ * @param email user's email
+ * @param password user's password
+ */
+const authenticate = async (email: string, password: string, remember: boolean = false): Promise<string | BoomType> => {
+    const repository: Repository<User> = getRepository(User)
+    const user: User | undefined = await repository.findOne({ where: { email }})
+
+    if (!user) {
+        return Boom.badRequest('user_not_found')
+    }
+
+    const passwordMatch = await bcrypt.compare(password, user.password)
+
+    if (!passwordMatch) {
+        return Boom.badRequest('incorrect_password')
+    }
+
+    delete user.password
+
+    const today = DateTime.local().toSeconds()
+    const dayInSeconds = 86400
+
+    return jwt.sign({
+        ...user,
+        exp: remember ? today + (30 * dayInSeconds): today + dayInSeconds,
+    }, secretKey)
+}
```

Comment:
Can you explain me

Reply:
Exp(iration) must be seconds. For non-remember sessions I put 1 day, for the remember sessions I put a month